app/sprinkles/account/src/Authenticate/PasswordSecurity.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * UserFrosting (http://www.userfrosting.com)
+ *
+ * @link      https://github.com/userfrosting/UserFrosting
+ * @copyright Copyright (c) 2019 Alexander Weissman
+ * @license   https://github.com/userfrosting/UserFrosting/blob/master/LICENSE.md (MIT License)
+ */
+
+namespace UserFrosting\Sprinkle\Account\Authenticate;
+
+use Illuminate\Cache\Repository as Cache;
+use UserFrosting\Support\Repository\Repository as Config;
+
+/**
+ * Handles enhanced password security methods for integration with Have I Been Pwned.
+ *
+ * @see https://haveibeenpwned.com/API/v2
+ * @author Amos Folz
+ */
+class PasswordSecurity
+{
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var Cache
+     */
+    protected $cache;
+
+    /**
+     * Create a new PasswordSecurity object.
+     *
+     * @param Cache  $cache  Cache service instance
+     * @param Config $config Config object that contains security settings.
+     */
+    public function __construct(Cache $cache, Config $config)
+    {
+        $this->cache = $cache;
+        $this->config = $config;
+    }
+
+    /**
+     * Generates a list of compromised passwords stored in cache or by querying Have I Been Pwned API.
+     *
+     * First check the cache to see if the hash prefix is stored.
+     * If not found in cache, query Have I Been Pwned API and store response in cache.
+     * @param  string $password
+     * @return string A numeric string representing the number of times a password has been compromised.
+     */
+    public function checkPassword($password)
+    {
+        // Get the SHA1 hash of our password.
+        // The first 5 characters (hash prefix) are sent to Have I Been Pwned.
+        $passwordHash = strtoupper(sha1($password));
+        $hashPrefix = substr($passwordHash, 0, 5);
+
+        $cacheMinutes = $this->config['site.password_security.enforce_no_compromised.cache'];
+
+        // We can use the comparison list directly from cache if it is found. Otherwise, we query Have I Been Pwned API and then save response to cache.
+        $hashArray = $this->cache->remember($hashPrefix, $cacheMinutes, function () use ($hashPrefix) {
+            return $this->getHashArrayFromAPI($hashPrefix);
+        });
+
+        return $this->checkHash($passwordHash, $hashArray);
+    }
+
+    /**
+     * Checks the maximum number of times that is acceptable for a password to have appeared in breaches.
+     *
+     * @return string Numeric string with -1 meaning disabled.
```

Comment:
Shouldn't this technically be an integer? Would be worth casting it as such too.

Reply:
There is no specific reason other than being a string by 'default'. I will update it to an integer.